internal/impl/kafka/enterprise/integration_test.go

```diff
@@ -1343,11 +1351,177 @@ output:
+		// We need to manually trigger a commit before closing the client because the default is to autocommit every 5s
+		err := client.CommitUncommittedOffsets(context.Background())
+		require.NoError(t, err)
+		client.Close()
+	}()
+
+	ctx, cancel := context.WithTimeout(context.Background(), 1*time.Second)
+	defer cancel()
+	fetches := client.PollRecords(ctx, count)
+	require.False(t, fetches.IsClientClosed())
+	err = fetches.Err()
+	if err != nil {
+		// If the context was cancelled, the producer finished so we won't get any more messages
+		if err != context.DeadlineExceeded {
+			require.NoError(t, err)
+		}
+		return nil
+	}
+
+	it := fetches.RecordIter()
+
+	if it.Done() {
+		return nil
+	}
+
+	var keys []int
+	for !it.Done() {
+		rec := it.Next()
+		key, err := strconv.Atoi(string(rec.Key))
+		require.NoError(t, err)
+		keys = append(keys, key)
+	}
+	return keys
+}
+
+// TestRedpandaMigratorConsumerGroupConsistencyIntegration checks that the consumer group updates are propagated
+// correctly when a consumer is switched from source to destination such that the destination consumer doesn't miss any
+// messages (even if it receives duplicates).
+func TestRedpandaMigratorConsumerGroupConsistencyIntegration(t *testing.T) {
+	integration.CheckSkip(t)
+	t.Parallel()
+
+	pool, err := dockertest.NewPool("")
+	require.NoError(t, err)
+	pool.MaxWait = time.Minute
+
+	source, err := startRedpanda(t, pool, true, false)
+	require.NoError(t, err)
+
+	destination, err := startRedpanda(t, pool, true, false)
+	require.NoError(t, err)
+
+	t.Logf("Source broker: %s", source.brokerAddr)
+	t.Logf("Destination broker: %s", destination.brokerAddr)
+
+	// Create the topic
+	dummyTopic := "foobar"
+	dummyRetentionTime := strconv.Itoa(int((1 * time.Hour).Milliseconds()))
+	createTopicWithACLs(t, source.brokerAddr, dummyTopic, dummyRetentionTime, "User:redpanda", kmsg.ACLOperationAll)
+
+	// Create a schema associated with the test topic
+	createSchema(t, source.schemaRegistryURL, dummyTopic, fmt.Sprintf(`{"name":"%s", "type": "record", "fields":[{"name":"test", "type": "string"}]}`, dummyTopic), nil)
+
+	dummyMessage := `{"test":"foo"}`
+	go func() {
+		t.Log("Producing messages...")
+		produceMessages(t, source, dummyTopic, dummyMessage, 0, 100, false, 50*time.Millisecond)
+
+		// Allow consumers a bit of time to read all the data
+		time.Sleep(100 * time.Millisecond)
+
+		t.Log("Finished producing messages")
+	}()
+
+	// Run the Redpanda Migrator bundle
+	runMigratorBundle(t, source, destination, dummyTopic, true, nil)
+	t.Log("Migrator started")
+
+	// Wait for a few records to be produced
+	time.Sleep(1 * time.Second)
+
+	// Fetch the first record key from the source to create the consumer group
+	dummyConsumerGroup := "foobar_cg"
+	keys := fetchRecordKeys(t, source.brokerAddr, dummyTopic, dummyConsumerGroup, 5)
+	require.Len(t, keys, 5)
+	require.Equal(t, 1, keys[0])
+
+	// Wait for the topic and consumer group to be replicated in the destination first
+	for {
```

Comment:
Can we add a condition (even if long) so we don't loop forever? Is something like 1 minute a good enough bound here?

Reply:
That got me to experiment with `require.Eventually` and it's a lot cleaner that way, thank you! :)